src/test/java/com/cloudbees/jenkins/support/util/WordReplacerTest.java

```diff
@@ -0,0 +1,154 @@
+import java.util.stream.Stream;
+
+public class WordReplacerTest {
+    @Test
+    public void wordReplacerTest() {
+        //The ignore case doesn't work in these cases:
+        //Chinese numbers lower: 一, 二,三,四,五,六,七,八,九,十,百,千,萬
+        //Chinese numbers upper: 壹,貳,參,肆,伍,陸,柒,捌,玖,拾,佰,仟,萬
+
+        String input =    "abcdefghijklmnñopqrstuvwxyz Beijing: 佩 One: 一, CasaBlanca: الدار البيضاء Casa Blanca. Numbers: 一, 二,三,四,五,六,七,八,九,十,百,千,萬";
+        String expected = "abcdefghijklmnñopqrstuvwxyz *beijing*: *佩* One: 一, CasaBlanca: *الدار البيضاء* Casa *BLANCA*. Numbers: 一, 二,三,四,五,六,七,八,九,十,百,千,萬";
+        // In an ideal world:
+        // String expected = "abcdefghijklmnñopqrstuvwxyz *beijing*: *佩* One: *1*, CasaBlanca: *الدار البيضاء* Casa *BLANCA*. Numbers: *1*, 二,三,四,五,六,七,八,九,十,百,千,萬";
+
+        String[] search = new String[] {"a", "Ñ", "beijing", "佩", "الدار البيضاء", "blaNca", "壹"};
+        String[] replace = new String[] {"*a*", "*ñ*", "*beijing*", "*佩*", "*الدار البيضاء*", "*BLANCA*", "*1*"};
+
+
+        String result = WordReplacer.replaceWordsIgnoreCase(input, search, replace);
+        System.out.format("%s ->\n%s", input, result);
+        Assert.assertEquals(expected, result);
+    }
+
+    @Test
+    public void sameResultTest() {
+        String[] searches = new String[]{"one", "two"};
+        String[] replaces = new String[]{"111", "111"};
+
+        // Populate the content mapping objects to filter
+        ContentMapping[] contentMappings = getContentMappings(searches, replaces);
+
+        // Texts to test against
+        // regexp doesn't replace ;word, we do. It's even better but we can't check it
+        String[] inputs = new String[] {
+                "", "a", "none", "one should be replaced", "onecar shouldn't be replaced", "two is replaced",
+                "must replace two", "shouldn't replace twoo", "one!one,one!onetwoone_twoone#one?one:one->one\"one",
+                "'one|twoone, \"one\\two one\ntwo one\ttwo one=two (one+two*one) all replaced ",
+                "one\\two one\ntwo one\ttwo one=two (one+two*one) all replaced "};
+
+        // For each text, filter using ContentMapping and with WordReplacer as well
+        for (String input : inputs) {
+            String resultCM = input;
+            for (ContentMapping cm : contentMappings) {
+                resultCM = cm.filter(resultCM);
+            }
+
+            String resultWordReplacer = WordReplacer.replaceWordsIgnoreCase(input, searches, replaces);
+
+            // Both filtered strings should be the same
+            Assert.assertEquals("The string replaced should be the same using RegExp or WordReplacer", resultCM, resultWordReplacer);
+        }
+    }
+
+   @Ignore("It was useful to make the decision to move out of Reg Exp. As the implementation of the Content Mapping is" +
+            "now WordReplacer, it has no sense to test. We keep it here for future cases.")
+    @Test
+    public void performanceTest() {
```

Comment:
This would be better handled using something like [JMH](https://openjdk.java.net/projects/code-tools/jmh/).

Reply:
I saw it but all examples measure a full method, with annotations and I wanted to measure specific pieces of code inside methods, so I ended up doing it by myself